packages/@aws-cdk/aws-codedeploy/lib/deployment-configuration.ts

```diff
@@ -0,0 +1,155 @@
+import cdk = require('@aws-cdk/cdk');
+import { cloudformation } from './codedeploy.generated';
+
+export class DeploymentConfigurationName extends cdk.CloudFormationToken {}
+
+export class DeploymentConfigurationArn extends cdk.Arn {}
+
+/**
+ * Represents a Deployment Configuration for an EC2/on-premise Deployment Group.
+ */
+export interface ServerDeploymentConfiguration {
+    readonly deploymentConfigurationName: DeploymentConfigurationName;
+    readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+}
+
+/**
+ * Pre-defined Deployment Configurations for an EC2/on-premise Deployment Group.
+ *
+ * @see https://docs.aws.amazon.com/codedeploy/latest/userguide/deployment-configurations.html#deployment-configuration-server
+ * @see #OneAtATime
+ * @see #HalfAtATime
+ * @see #AllAtOnce
+ */
+export class PredefinedServerDeploymentConfiguration implements ServerDeploymentConfiguration {
+    public static readonly OneAtATime = new PredefinedServerDeploymentConfiguration('CodeDeployDefault.OneAtATime');
```

Comment:
Intuitively, I'd expect these to be available on `DeploymentConfig`:

```
DeploymentConfig.OneAtAtTime
DeploymentConfig.AllAtOnes
DeploymentConfig.AllAtOnces
```

Reply:
I agree that `IServerDeploymentConfig` is ugly. Why isn't `ServerDeploymentConfig` an abstract base class?